src/store/active.js

```diff
@@ -1,57 +1,44 @@
-import {
-  SET_ACTIVE_MEDIA_ITEM,
-  PAUSE_ACTIVE_MEDIA_ITEM,
-  EJECT_ACTIVE_MEDIA_ITEM,
-  SET_MESSAGE,
-} from '~/constants/mutation-types'
+import { defineStore } from 'pinia'
+
+export const ACTIVE = 'active'
 
 /**
- * Stores information about the active media item.
- * @return {import('./types').ActiveMediaState}
+ * Store information about the active media item.
  */
-const state = () => ({
-  type: null,
-  id: null,
-  status: 'ejected', // can be 'playing' or 'paused' as well
-  message: null,
-})
+export const useActive = defineStore(ACTIVE, {
+  /** @return {import('./types').ActiveMediaState} */
+  state: () => ({
+    type: null,
+    id: null,
+    status: 'ejected', // 'ejected' means player is closed
+    message: null,
+  }),
 
-const mutations = {
-  /**
-   * Set the active media item.
-   * @param {import('./types').ActiveMediaState} _state
-   * @param {object} payload
-   * @param {'image' | 'audio' | null} payload.type - the nature of the active media item
-   * @param {string} payload.id - the ID of the active media item
-   * @param {'ejected' | 'playing' | 'paused'} payload.status - the status of the active media item
-   */
-  [SET_ACTIVE_MEDIA_ITEM](_state, { type, id, status = 'playing' }) {
-    _state.type = type
-    _state.id = id
-    _state.status = status
-  },
-  /**
-   * Pause the active media item.
-   * @param {import('./types').ActiveMediaState} _state
-   */
-  [PAUSE_ACTIVE_MEDIA_ITEM](_state) {
-    _state.status = 'paused'
-  },
-  /**
-   * Eject, and unset, the active media item.
-   * @param {import('./types').ActiveMediaState} _state
-   */
-  [EJECT_ACTIVE_MEDIA_ITEM](_state) {
-    _state.type = null
-    _state.id = null
-    _state.status = 'ejected'
+  actions: {
+    /**
+     * @param {object} payload
+     * @param {'image' | 'audio' | null} payload.type
```

Comment:
Should this really be allowed to be `null`?

Reply:
Yeah, when no media is active, this is set to `null`.

Reply:
But the _payload_ type shouldn't allow passing in `null`. The store value can be null for sure but there's no reason this function should allow it.

Reply:
You're right, I missed the fact that we have the `ejectActiveMediaItem` mutation specifically for this.